pallets/foreign-investments/src/tests.rs

```diff
@@ -0,0 +1,328 @@
+use cfg_traits::{investments::ForeignInvestment as ForeignInvestmentT, StatusNotificationHook};
+use cfg_types::investments::ForeignInvestmentInfo as ForeignInvestmentInfoS;
+use frame_support::assert_ok;
+
+use crate::{
+	hooks::FulfilledSwapOrderHook,
+	mock::*,
+	types::{InvestState, TokenSwapReason},
+	*,
+};
+
+const USER: AccountId = 1;
+const INVESTMENT_ID: InvestmentId = InvestmentId(42, 23);
+const USER_CURR: CurrencyId = 5;
+const POOL_CURR: CurrencyId = 10;
+const ORDER_ID: OrderId = 1;
+
+mod util {
+	use super::*;
+
+	pub fn new_invest(order_id: OrderId, amount: Balance) {
+		MockInvestment::mock_investment_requires_collect(|_, _| false);
+		MockInvestment::mock_investment(|_, _| Ok(0));
+		MockInvestment::mock_update_investment(|_, _, _| Ok(()));
+		MockTokenSwaps::mock_place_order(move |_, _, _, _, _, _| Ok(order_id));
+		MockCurrencyConversion::mock_stable_to_stable(move |_, _, _| Ok(amount) /* 1:1 */);
+
+		ForeignInvestment::increase_foreign_investment(
+			&USER,
+			INVESTMENT_ID,
+			amount,
+			USER_CURR,
+			POOL_CURR,
+		)
+		.unwrap();
+
+		MockInvestment::mock_investment_requires_collect(|_, _| unimplemented!("no mock"));
+		MockInvestment::mock_investment(|_, _| unimplemented!("no mock"));
+		MockInvestment::mock_update_investment(|_, _, _| unimplemented!("no mock"));
+		MockTokenSwaps::mock_place_order(|_, _, _, _, _, _| unimplemented!("no mock"));
+		MockCurrencyConversion::mock_stable_to_stable(|_, _, _| unimplemented!("no mock"));
+	}
+
+	pub fn notify_swaped(order_id: OrderId, amount: Balance) {
+		MockInvestment::mock_investment_requires_collect(|_, _| false);
+		MockInvestment::mock_investment(|_, _| Ok(0));
+		MockInvestment::mock_update_investment(|_, _, _| Ok(()));
+		MockTokenSwaps::mock_cancel_order(|_| Ok(()));
+		MockTokenSwaps::mock_is_active(|_| true);
+		MockCurrencyConversion::mock_stable_to_stable(move |_, _, _| Ok(amount) /* 1:1 */);
+
+		FulfilledSwapOrderHook::<Runtime>::notify_status_change(
+			order_id,
+			Swap {
+				currency_out: USER_CURR,
+				currency_in: POOL_CURR,
+				amount: amount,
```

Comment:
```suggestion
				amount,
```

Also, _super_ `nit` but for me it would read better by ordering these fields more narratively: `currency_in, `amount_in`, currency_out`.

Reply:
I think it's `amount_out`, but I agree with your point 👍🏻

Reply:
The amount of a swap should always be the `amount_in` inherited from the `pallet-orderbook` where an order is placed with the `amount_in`. There have been some discussions around this, but this was the decision.

Reply:
Just to confirm. From the docs of `Swap` type, currently there are the following commented lines:
```rust
	/// The incoming currency, i.e. the desired one.
	pub currency_in: Currency,
	/// The outgoing currency, i.e. the one which should be replaced.
	pub currency_out: Currency,
	/// The amount of outgoing currency which shall be exchanged.
	pub amount: Balance,
```
Seems like `amount` is the `amount_out` (?)

Reply:
Ouch, documentation bug! It should definitely be the amount of incoming currency. Will fix soon. I assumed the swap amounts to be outgoing currency in the early implementation and then realised that the `OrderBook` requires `buy_amount` in incoming currency which we need to follow in foreign ivnestments.

Here's the conversation in the DAO Slack as reference: https://centrifugedao.slack.com/archives/C04HCK98KLL/p1693205602696709